server/src/collection.rs

```diff
@@ -156,4 +152,26 @@ impl Collection {
     pub async fn alias(&self) -> String {
         self.alias.lock().await.clone()
     }
+
+    pub async fn dispatch_items(&self, object_server: &zbus::ObjectServer) -> Result<(), Error> {
+        let keyring_items = self.keyring.items().await;
+        let mut n_items: u32 = 1;
+
+        for keyring_item in keyring_items {
+            let item = item::Item::new(
+                keyring_item.map_err(Error::InvalidItem)?,
+                self.is_locked().await,
+                Arc::clone(&self.manager),
+                self.path.clone(),
+                n_items,
+            );
+            *self.item_index.write().await = n_items;
+            n_items += 1;
+
+            self.items.lock().await.push(item.path().clone());
```

Comment:
Get the lock once, before the for iterations.

Reply:
Done.